cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +119,25 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	// validate if a federation namespace is configured
+	if len(opts.Config.FederationNamespace) == 0 {
+		klog.Fatalf("The federation namespace must be specified via --federation-namespace")
+	}
+
+	if len(masterURL) == 0 && len(kubeconfig) == 0 && len(opts.Config.FederationNamespace) == 0 {
```

Comment:
(No action required) It's often helpful to use a boolean variable to document your intent with a conditional, e.g.

```golang
runningInCluster := len(masterURL) == 0 && len(kubeconfig) == 0
if runningInCluster && len(opts.Config.FederationNamespace) == 0 {
 ...
}
```

Reply:
I think I like your rationale for clubbing two conditions here and mentioning the exact intent behind it. I will accommodate this change in this PR.